### PR TITLE
fix(components): [tabs] nav offset error when use both number and string

### PR DESCRIPTION
**Pull request by tolking**
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

Wrong nav offset when there have a number and a string the same.

[before](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBsYW5nPVwidHNcIiBzZXR1cD5cbmltcG9ydCB7IHJlZiB9IGZyb20gJ3Z1ZSdcblxuY29uc3QgYWN0aXZlTmFtZSA9IHJlZignMicpXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8ZWwtdGFicyB2LW1vZGVsPVwiYWN0aXZlTmFtZVwiPlxuICAgIDxlbC10YWItcGFuZSBsYWJlbD1cIlN0cmluZyAxXCIgbmFtZT1cIjFcIj5Sb2xlPC9lbC10YWItcGFuZT5cbiAgICA8ZWwtdGFiLXBhbmUgbGFiZWw9XCJTdHJpbmcgMlwiIG5hbWU9XCIyXCI+VXNlcjwvZWwtdGFiLXBhbmU+XG4gICAgPGVsLXRhYi1wYW5lIGxhYmVsPVwiTnVtYmVyIDJcIiA6bmFtZT1cIjJcIj5Db25maWc8L2VsLXRhYi1wYW5lPlxuICAgIDxlbC10YWItcGFuZSBsYWJlbD1cIlN0cmluZyAzXCIgbmFtZT1cIjNcIj5UYXNrPC9lbC10YWItcGFuZT5cbiAgPC9lbC10YWJzPlxuPC90ZW1wbGF0ZT5cbiIsImltcG9ydF9tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHt9XG59IiwiX28iOnt9fQ==)

[after](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBsYW5nPVwidHNcIiBzZXR1cD5cbmltcG9ydCB7IHJlZiB9IGZyb20gJ3Z1ZSdcblxuY29uc3QgYWN0aXZlTmFtZSA9IHJlZignMicpXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8ZWwtdGFicyB2LW1vZGVsPVwiYWN0aXZlTmFtZVwiPlxuICAgIDxlbC10YWItcGFuZSBsYWJlbD1cIlN0cmluZyAxXCIgbmFtZT1cIjFcIj5Sb2xlPC9lbC10YWItcGFuZT5cbiAgICA8ZWwtdGFiLXBhbmUgbGFiZWw9XCJTdHJpbmcgMlwiIG5hbWU9XCIyXCI+VXNlcjwvZWwtdGFiLXBhbmU+XG4gICAgPGVsLXRhYi1wYW5lIGxhYmVsPVwiTnVtYmVyIDJcIiA6bmFtZT1cIjJcIj5Db25maWc8L2VsLXRhYi1wYW5lPlxuICAgIDxlbC10YWItcGFuZSBsYWJlbD1cIlN0cmluZyAzXCIgbmFtZT1cIjNcIj5UYXNrPC9lbC10YWItcGFuZT5cbiAgPC9lbC10YWJzPlxuPC90ZW1wbGF0ZT5cbiIsIlBsYXlncm91bmRNYWluLnZ1ZSI6IjxzY3JpcHQgc2V0dXA+XG5pbXBvcnQgQXBwIGZyb20gJy4vQXBwLnZ1ZSdcbmltcG9ydCB7IHNldHVwRWxlbWVudFBsdXMgfSBmcm9tICcuL2VsZW1lbnQtcGx1cy5qcydcbnNldHVwRWxlbWVudFBsdXMoKVxuPC9zY3JpcHQ+XG5cbjx0ZW1wbGF0ZT5cbiAgPEFwcCAvPlxuPC90ZW1wbGF0ZT5cbiIsImltcG9ydF9tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHtcbiAgICBcImVsZW1lbnQtcGx1c1wiOiBcImh0dHBzOi8vcHJldmlldy04NTY1LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguZnVsbC5taW4ubWpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXMvXCI6IFwidW5zdXBwb3J0ZWRcIlxuICB9XG59IiwiaW1wb3J0LW1hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge1xuICAgIFwidnVlXCI6IFwiaHR0cHM6Ly91bnBrZy5jb20vQHZ1ZS9ydW50aW1lLWRvbUBsYXRlc3QvZGlzdC9ydW50aW1lLWRvbS5lc20tYnJvd3Nlci5qc1wiLFxuICAgIFwiQHZ1ZS9zaGFyZWRcIjogXCJodHRwczovL3VucGtnLmNvbS9AdnVlL3NoYXJlZEBsYXRlc3QvZGlzdC9zaGFyZWQuZXNtLWJ1bmRsZXIuanNcIixcbiAgICBcImVsZW1lbnQtcGx1c1wiOiBcImh0dHBzOi8vcHJldmlldy04NTY1LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguZnVsbC5taW4ubWpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXMvXCI6IFwidW5zdXBwb3J0ZWRcIixcbiAgICBcIkBlbGVtZW50LXBsdXMvaWNvbnMtdnVlXCI6IFwiaHR0cHM6Ly91bnBrZy5jb20vQGVsZW1lbnQtcGx1cy9pY29ucy12dWVAMi9kaXN0L2luZGV4Lm1pbi5qc1wiXG4gIH0sXG4gIFwic2NvcGVzXCI6IHt9XG59IiwiZWxlbWVudC1wbHVzLmpzIjoiaW1wb3J0IHsgZ2V0Q3VycmVudEluc3RhbmNlIH0gZnJvbSAndnVlJ1xuaW1wb3J0IEVsZW1lbnRQbHVzIGZyb20gJ2VsZW1lbnQtcGx1cydcblxubGV0IGluc3RhbGxlZCA9IGZhbHNlXG5hd2FpdCBsb2FkU3R5bGUoKVxuXG5leHBvcnQgZnVuY3Rpb24gc2V0dXBFbGVtZW50UGx1cygpIHtcbiAgaWYgKGluc3RhbGxlZCkgcmV0dXJuXG4gIGNvbnN0IGluc3RhbmNlID0gZ2V0Q3VycmVudEluc3RhbmNlKClcbiAgaW5zdGFuY2UuYXBwQ29udGV4dC5hcHAudXNlKEVsZW1lbnRQbHVzKVxuICBpbnN0YWxsZWQgPSB0cnVlXG59XG5cbmV4cG9ydCBmdW5jdGlvbiBsb2FkU3R5bGUoKSB7XG4gIHJldHVybiBuZXcgUHJvbWlzZSgocmVzb2x2ZSwgcmVqZWN0KSA9PiB7XG4gICAgY29uc3QgbGluayA9IGRvY3VtZW50LmNyZWF0ZUVsZW1lbnQoJ2xpbmsnKVxuICAgIGxpbmsucmVsID0gJ3N0eWxlc2hlZXQnXG4gICAgbGluay5ocmVmID0gJ2h0dHBzOi8vcHJldmlldy04NTY1LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguY3NzJ1xuICAgIGxpbmsuYWRkRXZlbnRMaXN0ZW5lcignbG9hZCcsIHJlc29sdmUpXG4gICAgbGluay5hZGRFdmVudExpc3RlbmVyKCdlcnJvcicsIHJlamVjdClcbiAgICBkb2N1bWVudC5ib2R5LmFwcGVuZChsaW5rKVxuICB9KVxufSIsIl9vIjp7InNob3dIaWRkZW4iOnRydWUsInN0eWxlU291cmNlIjoiaHR0cHM6Ly9wcmV2aWV3LTg1NjUtZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9pbmRleC5jc3MifX0=)

### CODE REVIEW

File: packages/components/tabs/src/tab-pane.vue [hunk 1]
```diff
@@ -2,15 +2,16 @@
   <div
     v-if="shouldBeRender"
     v-show="active"
-    :id="`pane-${paneName}`"
+    :id="`pane-${isNumber(paneName) ? 'n-' : ''}${paneName}`"
     :class="ns.b()"
     role="tabpanel"
     :aria-hidden="!active"
-    :aria-labelledby="`tab-${paneName}`"
+    :aria-labelledby="`tab-${isNumber(paneName) ? 'n-' : ''}${paneName}`"
```
Comment (buqiyuan) [inline packages/components/tabs/src/tab-pane.vue:9]: I found that there are many similar codes, can we encapsulate it into a general function?
like this:
```ts 
const genId = (type: 'tab' | 'pane', name: string) => {
  return `${type}-${isNumber(name) ? 'n-' : ''}${name}`
}
```

Comment (tolking) [reply packages/components/tabs/src/tab-pane.vue]: @buqiyuan I refactored the relevant code

Comment (buqiyuan) [reply packages/components/tabs/src/tab-pane.vue]: /cc @element-plus/backers  Should we be compatible with string and number types

**Comment by buqiyuan** [comment]
@tolking /cc #8466

**Review by buqiyuan — APPROVED** [review]

**Review by YunYouJun — APPROVED** [review]

File: packages/components/tabs/src/tab-pane.vue (hunk 1 above)
Comment (YunYouJun) [reply packages/components/tabs/src/tab-pane.vue]: > Should we be compatible with string and number types

In my opinion, they should be string at last, right?

Comment (tolking) [reply packages/components/tabs/src/tab-pane.vue]: Exactly so, on [line 44](https://github.com/element-plus/element-plus/blob/dev/packages/components/tabs/src/tab-bar.vue#L44) will not be able to distinguish they.

Comment (YunYouJun) [reply packages/components/tabs/src/tab-pane.vue]: Do we need to distinguish between them?

Comment (tolking) [reply packages/components/tabs/src/tab-pane.vue]: 0 -> 'tab-0'
'0' -> 'tab-0'
The id and key need to be unique values, so there are some potential risks.

Of course, it is also possible not to distinguish between them. This issues can also be fixed by class name, but I am not sure if it is a good idea.

**Pull request closed without merging**